app/controllers/concerns/json_collection_pagination.rb

```diff
@@ -4,56 +4,47 @@ module JsonCollectionPagination
   SERIALIZER = nil
 
   def paginated_json_collection(collection, path_helper, sort_key, sort_order, offset, per_page, total, params)
-    #changing from order to reorder to discard previous ordering
-    sort_key = (sort_key.presence || :id).to_sym
-    sort_order = sort_order.to_s == 'desc' ? :desc : :asc
-    fail ArgumentError.new('Invalid sort_key') unless collection.column_names.include?(sort_key.to_s)
-    collection = collection.reorder(sort_key => sort_order)
+    # Backwards compatibility
+    params[:sort_key] = sort_key
+    params[:sort_order] = sort_order
+    params[:offset] = offset
+    params[:per_page] = per_page
+    params[:total] = total
+    self.params = params
+    paginated_json_collection_new(collection)
+  end
 
+  def paginated_json_collection_new(collection)
```

Comment:
Or simpler in the long run to just go through and update all the existing calls to `paginated_json_collection`?